kbase-extension/static/kbase/js/widgets/narrative_core/catalog/kbaseCatalogBrowser.js

```diff
@@ -0,0 +1,1105 @@
+                    .then(function (modules) {
+                        for(var k=0; k<modules.length; k++) {
+                            var m = {
+                                info: modules[k],
+                                $div: $('<div>').addClass('kbcb-module')
+                            };
+                            self.renderModuleBox(m);
+                            self.moduleList.push(m);
+                        }
+                    })
+                    // [NARRATIVE_EDIT] - catch to fail
+                    .fail(function (err) {
+                        console.error('ERROR');
+                        console.error(err);
+                    });
+            },
+
+            updateRunStats: function() {
+                var self = this
+
+                var options = {};
+
+                return self.catalog.get_exec_aggr_stats(options)
+                    .then(function (stats) {
+                        self.runStats = stats;
+                        for(var k=0; k<stats.length; k++) {
+
+                            var lookup = stats[k].full_app_id;
+                            var idTokens = stats[k].full_app_id.split('/');
+                            if( idTokens.length === 2) {
+                                lookup = idTokens[0].toLowerCase() + '/' + idTokens[1];
+                            }
+                            if(self.appLookup[lookup]) {
+                                self.appLookup[lookup].setRunCount(stats[k].number_of_calls);
+                            }
+                        }
+                    })
+                    // [NARRATIVE_EDIT] - catch to fail
+                    .fail(function (err) {
+                        console.error('ERROR');
+                        console.error(err);
+                    });
+            },
+
+
+            updateFavoritesCounts: function() {
+                var self = this
+                var listFavoritesParams = { };
+                return self.catalog.list_favorite_counts(listFavoritesParams)
+                    .then(function (counts) {
+                        for(var k=0; k<counts.length; k++) {
+                            var c = counts[k];
+                            var lookup = c.id;
+                            if(c.module_name_lc != 'nms.legacy') {
+                                lookup = c.module_name_lc + '/' + lookup
+                            }
+                            if(self.appLookup[lookup]) {
+                                self.appLookup[lookup].setStarCount(c.count);
+                            }
+                        }
+                    })
+                    // [NARRATIVE_EDIT] - catch to fail
+                    .fail(function (err) {
+                        console.error('ERROR');
+                        console.error(err);
+                    });
+            },
+
+            // warning!  will not return a promise if the user is not logged in!
```

Comment:
Thanks. I was wondering about how to handle that situation. I'll update that in the kbase-ui version of this.  In the Narrative version, that warning doesn't apply because right now you must be logged in.